### PR TITLE
docs: reviewed type-safe-api topic

### PR DESCRIPTION
**Pull request by swap-aws**
Fixes #

### CODE REVIEW

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (cogwirrel) [inline packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: Does `_question` render something differently in the docs website? :)

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (cogwirrel) [inline packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: Perhaps we get rid of the ", provided an integration (eg a lambda) exists for every operation" here? Integrations are explained later so it might just add confusion :)

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (cogwirrel) [inline packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: "...in your desired languages, _which_ you can use..."?

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (cogwirrel) [inline packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: Might make sense to make this an explicit numbered step, eg "Edit your `.projenrc` and configure TypeSafeApi"?

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (cogwirrel) [inline packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: I think "for your package manager" should also be in the brackets, since the commands are different for different package managers :)

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (cogwirrel) [inline packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: I think "in the CDK construct" is a bit confusing here :) Also the intention of the "we can" is more "it's possible to" since we're using a NodejsFunction, rather than creating a separate package for the lambda function code :). How about:

"In the TypeScript CDK application, we used `NodejsFunction` which allows you to write your lambda handler code in the same `infra` project. Define `say-hello.ts` and make use of the generated lambda handler wrapper."

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (cogwirrel) [inline packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: I'm not sure "to troubleshoot this" makes sense here - the error is expected because the CDK construct requires an integration for every operation, and we just added a new operation which we haven't added an integration for yet :)

Maybe it makes more sense to say:

> Since an integration must be defined for every operation, you may see a build error in your CDK application.

{{error}}

> This is expected, and can be resolved by continuing with the steps below."

**Review by cogwirrel — COMMENTED** [review]
Thanks for going through this Swapnil! Added a few comments :)

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (swap-aws) [reply packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: @cogwirrel - no, this was me trying to use conventional commits inside markdown. 😬

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (swap-aws) [reply packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: > "...in your desired languages, _which_ you can use..."?

updated

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (swap-aws) [reply packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: > Perhaps we get rid of the ", provided an integration (eg a lambda) exists for every operation" here? Integrations are explained later so it might just add confusion :)

removed

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (swap-aws) [reply packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: updated

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (swap-aws) [reply packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: updated

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (swap-aws) [reply packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: updated

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (swap-aws) [reply packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: updated

**Review by swap-aws — COMMENTED** [review]
Updated from feedback

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (cogwirrel) [reply packages/type-safe-api/docs/developer_guides/type-safe-api/index.md]: ooh we probably still want to delete this before we merge :) otherwise looks great! 😄

**Review by cogwirrel — APPROVED** [review]

File: packages/type-safe-api/docs/developer_guides/type-safe-api/index.md
Comment (cogwirrel) [inline packages/type-safe-api/docs/developer_guides/type-safe-api/index.md:11]: Ooh do you know if this (and the other `!!!` directives) will render differently in the UI? Might be worth building locally and taking a look at the generated html documentation if you haven't already :)

**Comment by cogwirrel** [comment]
Looks great! I think the only thing would be to check that the `!!!note` etc syntax works ok for the docs website, and if not perhaps consider a different way to represent them? Otherwise happy to ship! :)

**Comment by swap-aws** [comment]
@cogwirrel, @agdimech tried it out and it works fine :)

**Comment by cogwirrel** [comment]
> @cogwirrel, @agdimech tried it out and it works fine :)

Fantastic! 😄

**Review by cogwirrel — APPROVED** [review]

**Comment by swap-aws** [comment]
@cogwirrel I don't think I have merge rights, so feel free to merge if needed.

**Pull request merged**
